### ISSUE TITLE
missing_docs not triggered when missing docs on trait alias

### ISSUE THREAD

**Issue by chloekek** [rust-lang/rust#111025]
I tried this code:

```rust
//! example
#![feature(trait_alias)]
#![warn(missing_docs)]
pub trait T = Send;
```

I expected to see this happen: “missing documentation for a [trait alias]” diagnostic

Instead, this happened: no diagnostic

### Meta

Using 1.71.0-nightly 2023-04-29 87b1f891ea76713462cf.

[trait alias]: https://github.com/rust-lang/rust/issues/55628

**Comment by mj10021** [comment]
Running the following code on the nightly:
``` rust
#![feature(trait_alias)]
#![warn(missing_docs)]

pub trait T = Send;
```

gives the following output:
``` rust
   Compiling playground v0.0.1 (/playground)
warning: missing documentation for the crate
 --> src/lib.rs:1:1
  |
1 | / #![feature(trait_alias)]
2 | | #![warn(missing_docs)]
3 | |
4 | | pub trait T = Send;
  | |___________________^
  |
note: the lint level is defined here
 --> src/lib.rs:2:9
  |
2 | #![warn(missing_docs)]
  |         ^^^^^^^^^^^^
  ```
  
  So I guess the difference is the `//! example` ?

**Comment by mj10021** [comment]
I am trying to figure out if this is desired behavior or not based on rust/compiler/rustc_lint/src/builtins.rs but I am not sure exactly why the //! disables the lint, does it change the visibility of the trait alias?

**Comment by chloekek** [comment]
That diagnostic warns about missing docs on the crate, but not on the trait alias.

**Comment by mj10021** [comment]
> That diagnostic warns about missing docs on the crate, but not on the trait alias.

Ah yes, thanks

**Comment by chloekek** [comment]
This is now fixed.
